engine/base/kafka/src/main/scala/pl/touk/nussknacker/engine/baseengine/kafka/LoopUntilClosed.scala

```diff
@@ -4,32 +4,30 @@ import com.typesafe.scalalogging.LazyLogging
 import org.apache.commons.lang3.concurrent.BasicThreadFactory
 import org.apache.kafka.common.errors.InterruptException
 
-import java.lang.Thread.UncaughtExceptionHandler
 import java.util.concurrent.atomic.AtomicBoolean
-import java.util.concurrent.{Executors, TimeUnit}
+import java.util.concurrent.{Callable, CompletableFuture, Executors, TimeUnit}
+import scala.concurrent.{ExecutionContext, Future}
 import scala.concurrent.duration.Duration
+import scala.compat.java8.FutureConverters._
 import scala.util.control.NonFatal
 
 //Runs task in loop, in several parallel copies restarting on errors
 //TODO: probably there is some util for that? :)
 class TaskRunner(taskName: String,
                  taskParallelCount: Int,
                  singleRun: String => Task,
-                 terminationTimeout: Duration,
-                 fatalErrorHandler: UncaughtExceptionHandler) extends AutoCloseable with LazyLogging {
+                 terminationTimeout: Duration) extends AutoCloseable with LazyLogging {
 
   private val threadFactory = new BasicThreadFactory.Builder()
     .namingPattern(s"worker-$taskName-%d")
-    .uncaughtExceptionHandler(fatalErrorHandler)
     .build()
 
   private val threadPool = Executors.newFixedThreadPool(taskParallelCount, threadFactory)
 
-  private val tasks = (0 until taskParallelCount).map(idx => new LoopUntilClosed(() => singleRun(s"task-$idx")))
+  private val tasks = (0 until taskParallelCount).map(idx => new LoopUntilClosed(() => singleRun(s"task-$idx"))).toList
```

Comment:
If we already in scala future, why not to just do `Future()` and use `ExecutionContext.fromExecutor`? Effect should be the same.

Reply:
:+1: